extension/indexer/tx_indexer.go

```diff
@@ -0,0 +1,118 @@
+// Copyright (C) 2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package indexer
+
+import (
+	"context"
+	"encoding/binary"
+	"errors"
+
+	"github.com/ava-labs/avalanchego/database"
+	"github.com/ava-labs/avalanchego/ids"
+
+	"github.com/ava-labs/hypersdk/chain"
+	"github.com/ava-labs/hypersdk/consts"
+	"github.com/ava-labs/hypersdk/fees"
+)
+
+var (
+	_ AcceptedSubscriber = (*txIndexer)(nil)
+	_ AcceptedSubscriber = (*noopTxIndexer)(nil)
+)
+
+var (
+	failureByte = byte(0x0)
+	successByte = byte(0x1)
+)
+
+type TxIndexer interface {
+	Accepted(ctx context.Context, blk *chain.StatelessBlock) error
+	GetTransaction(txID ids.ID) (bool, int64, bool, fees.Dimensions, uint64, error)
```

Comment:
Should we return a struct here instead? We can define an `indexer.Tx` struct type or something that the indexer defines for indexed tx data.

Reply:
This is an interface, so we can swap a noop version in if tx indexing is disabled as per https://github.com/ava-labs/hypersdk/pull/1143#discussion_r1679556902